src/Factory.php

```diff
@@ -107,7 +107,7 @@ public function create($spec): ElementInterface
         $spec = $this->validateSpecification($spec, __METHOD__);
         $type = $spec['type'] ?? Element::class;
 
-        $element = $this->getFormElementManager()->get($type);
+        $element = $this->getFormElementManager()->get($type, $spec['options'] ?? []);
```

Comment:
options must be passed as `['options' => $options]` according to element factory.

Passing non-null build options makes form manager do `build()` instead of `get()` that disregards service sharing options always resulting in new instance. I've seen forms used as single instance shared with form manager. FormElementManager is set to non-sharing by default however.

If this change is done it should be by passing null if options are empty or not set, not a fallback array.

Reply:
If element name is provided in the spec it should probably be passed along with the options for ElementFactory to pick up.